Feature/player movement

lizziebriggs opened the pull request:
Created a player movement script

lizziebriggs committed:
Started player movement

Created a player movement script

Banananaman91 committed:
Sorted .gitignore

I made a boo boo and fixed the .gitignore

lizziebriggs committed:
Completed player controls

Player now only moves along adjacent tiles which the user clicks on to move the player to that specific tile

Banananaman91 committed:
Merge remote-tracking branch 'origin/feature/Player-Movement' into feature/Player-Movement

Banananaman91 committed:
Changes to player controller

Change the way the player controller functions. Currently not fully working, needs more work.

Banananaman91 committed:
Begun PathFinding algorithm

Begun creating the PathFinding algorithm. Reorganised files and folders.

Banananaman91 committed:
Removed unused game objects from scene

patrickklaeren commented on GameProject/Assets/Turn based assets/Scripts/CubeGrower.cs:
```diff
@@ -0,0 +1,17 @@
+{
+    public void Select()
+    {
+        this.transform.localScale = this.transform.localScale * 3;
```
It'd probably be good to have a `const` for the size factor. Have it one place, change it in one place.

```cs
private const int ScaleFactor = 3;

public void Select()
{
    this.transform.localScale = this.transform.localScale * ScaleFactor;
}
```

Not sure if you guys have a naming convention for constants, just went for pascal here.

RubyNova replied:
I think this might have been done in another branch already, there's been a lot of duped work amongst a couple of branches

patrickklaeren commented on GameProject/Assets/Turn based assets/Scripts/CubeGrower.cs:
```diff
@@ -0,0 +1,17 @@
+
+    public void DeSelect()
+    {
+        this.transform.localScale = this.transform.localScale / 3;
```
Once you've introduced a constant, you can replace the magic `3` here.

patrickklaeren commented on GameProject/Assets/Turn based assets/Scripts/MouseSelection.cs:
```diff
@@ -0,0 +1,63 @@
+        public float distanceY;
+        public float offset;
+        public float gridSize;
+        public Plane _plane;
```
There's quite a few varying conventions going on here - I haven't seen a conventions file or anything pointing towards how you name public and private fields in this project, but switching between prefixing public fields and private fields with `_` seems confusing, maybe normalizing it would be a good start - especially for publicly exposed fields.

RubyNova replied:
Yeah this seems like another case of duped work

patrickklaeren commented on GameProject/Assets/Turn based assets/Scripts/MouseSelection.cs:
```diff
@@ -0,0 +1,63 @@
+                rawGridPoint += Vector3.one * offset;
+                player.FindPossibleMovePositions(rawGridPoint);
+                if(Input.GetKeyDown(KeyCode.Mouse0)) player.MovePlayer(rawGridPoint);
+                //if (player != null) player.transform.position = rawGridPoint;
```
If the code isn't needed, this could be removed?

patrickklaeren commented on GameProject/Assets/Turn based assets/Scripts/MouseSelection.cs:
```diff
@@ -0,0 +1,63 @@
+
+            if (_plane.Raycast(ray, out gridSpace))
+            {
+                rawGridPoint = ray.GetPoint(gridSpace);
```
This seems like an odd reuse of the `rawGridPoint` field - with the code commented out, it seems like the field doesn't need to be scoped to the behavior anymore and could be placed in scope of the method instead.

patrickklaeren commented on GameProject/Assets/Turn based assets/Scripts/MouseSelection.cs:
```diff
@@ -0,0 +1,63 @@
+            if(Input.GetKeyDown(KeyCode.Mouse0))
+            {
+                RaycastHit hit;
+                if (Physics.Raycast(ray, out hit))
```
This could be changed to `if(Physics.Raycast(ray, out RaycastHit hit)`, where the variable is initialized as part of the `out`.

patrickklaeren commented on GameProject/Assets/Turn based assets/Scripts/MouseSelection.cs:
```diff
@@ -0,0 +1,63 @@
+                }
+                else
+                {
+                    if (_selection != null) _selection.DeSelect();
```
Looks like this branch of code always runs, and could be simplified:
```cs
if(Input.GetKeyDown(KeyCode.Mouse0))
{
    _selection?.DeSelect();

    if (Physics.Raycast(ray, out RaycastHit hit))
    {
        _selection = hit.collider.GetComponent<ISelection>();
        _selection.Select();
    }
}
```

patrickklaeren commented on GameProject/Assets/Turn based assets/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,82 @@
+
+        private void Update()
+        {
+//            if(_findPossibleMoves)
```
Could do with removing the commented out code

patrickklaeren commented on GameProject/Assets/Turn based assets/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,82 @@
+
+
+        // Purely for fun
+        private void YeetThePlayer()
```
Liking the method naming

patrickklaeren commented on GameProject/Assets/Turn based assets/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,82 @@
+        // Purely for fun
+        private void YeetThePlayer()
+        {
+            if (_mouseSelection != null)
```
I would probably negate and short circuit here:
```cs
if(_mouseSelection == null)
    return;
```

RubyNova replied:
To add to this - if `MouseSelection` is not a `MonoBehaviour` or other in-game-world Unity object you could replace the `==` with `is`.

patrickklaeren commented on GameProject/Assets/Turn based assets/Scripts/SetPlayerPosition.cs:
```diff
@@ -0,0 +1,26 @@
+            if(playerControler != null)
+            {
+                playerControler.currentPos = transform.position;
+                //playerControler.MovePlayer();
```
Commented out code could be removed.

patrickklaeren commented on GameProject/Assets/Turn based assets/Scripts/SetPlayerPosition.cs:
```diff
@@ -0,0 +1,26 @@
+        {
+            PlayerController playerControler = FindObjectOfType<PlayerController>();
+
+            if(playerControler != null)
```
I would probably suggest negating and short circuiting this

patrickklaeren commented on GameProject/Assets/Turn based assets/Scripts/TileGenerator.cs:
```diff
@@ -0,0 +1,53 @@
+        GenerateTiles();
+    }
+
+    public void GenerateTiles()
```
Nitpick, but this could be `private` seeing as you've called it from `Start()`, which is good.

Banananaman91 committed:
Merging development into player movement

Banananaman91 committed:
New MouseSelection type from development branch

Banananaman91 committed:
Merged player-movement to pathfinding

Banananaman91 committed:
Removed duped types from merging

Banananaman91 committed:
Fixed errors from merging

Banananaman91 committed:
Bugfix endless loop

fixed bug in PathFinder that generated an endless loop.

lizziebriggs committed:
Minor tweak to path finding

Banananaman91 committed:
Fixes to pathfinder

fixed unity crash for pathfinder, turned it into a coroutine, and altered some syntax. Currently still does not move towards target. F values not working properly.

lizziebriggs committed:
Neatened code

Minor tweaks to the code to neaten it up and moved the calculation for the rawGridPoint to a seperate method. No new problems created.

lizziebriggs committed:
Fixed pathfinding algorithm

Fixed the pathfinding algorithm and began creating a method to create a path the player follows.

lizziebriggs committed:
Player moving cont.

Player can move but unity crashes when you try to move more than once

Banananaman91 committed:
Changes to pathFinder return value

Path Finder now returns list value for use, debug shows it working but player still isn't moving

lizziebriggs committed:
player movement cont.

Banananaman91 committed:
Cleaned up code

Cleaned up code, removed commented code. PathFinder appears to mostly generate paths to the right of the player (in scene)

Banananaman91 committed:
Added .gitkeep for empty folders

Banananaman91 committed:
Completed Path Finder

path finder finished, currently returns to player for testing and shows correct path drawn out with movement. Ready to be used for proper player movement and AI movement

Banananaman91 committed:
Final adjustments to Path Finder

Path finder has been adjusted to work for 2D and 3D path finding, to be used across turn based AI and free roaming AI

Banananaman91 committed:
Merge pull request #41 from Banananaman91/feature/pathfinding

Feature/pathfinding

DatPinkGuy committed:
Visible character movement & Precise waypoint selection

Added visible character movement to the waypoint, tile is created now in the position of mouse click, second click on tile commences the movement

Banananaman91 committed:
Merge pull request #43 from Banananaman91/feature/pathfinding

Visible character movement & Precise waypoint selection

Banananaman91 committed:
Put pathfinder behind interface

Created interface for pathfinding, created new messenger system for handling initialisation of pathfinder

Banananaman91 committed:
Merge pull request #44 from Banananaman91/feature/pathfinding

Put pathfinder behind interface

DatPinkGuy committed:
Player Model Placeholder, Turning animation

Added placeholder for player model, creating turning animation during movement to different positions

Banananaman91 committed:
Object avoidance for path finder

Path finding now takes in list of positions to be avoided based on objects in a radius. This also ignores the tile generated for the target position

DatPinkGuy committed:
Added visualized path to Target Position

Visualized path to target position which appears on mouse click and disappears upon reaching target position. Added material to add transparency to the path created.

Banananaman91 committed:
Merge branch 'feature/pathfinding' into feature/Player-Movement

Banananaman91 committed:
Merge pull request #45 from Banananaman91/feature/Player-Movement

Feature/player movement

Banananaman91 committed:
Created controller listener

Banananaman91 committed:
Fixed controller using pathfinder

Banananaman91 committed:
Created object positioner type

This type repositions standard objects to whole values

Banananaman91 committed:
Object avoidance working

Object avoiding fully works, path no longer draws through larger objects. Currently lacks efficiency, performance takes a big hit.

Banananaman91 committed:
Moved object avoidance from player to pathfinder

Pathfinder now contains object avoidance method, usable by all that use pathfinder

Banananaman91 committed:
Code review changes

Made appropriate changes based on code review

Banananaman91 committed:
Scene changes

Saved changes within unity scene

Banananaman91 committed:
Merge pull request #46 from Banananaman91/feature/pathfinding

Feature/pathfinding

Banananaman91 committed:
Minor change to using namespace

Banananaman91 committed:
Minor changes, added terrain to scene

Mouse Selection needs redesign to detect terrain that blocks plane, otherwise still being clickable if below

Banananaman91 committed:
Cleaned up code
